src/headingengine.js

```diff
@@ -71,21 +83,67 @@ export default class HeadingEngine extends Plugin {
 	afterInit() {
 		// If the enter command is added to the editor, alter its behavior.
 		// Enter at the end of a heading element should create a paragraph.
-
 		const editor = this.editor;
 		const command = editor.commands.get( 'heading' );
 		const enterCommand = editor.commands.get( 'enter' );
+		const options = this._getLocalizedOptions();
 
 		if ( enterCommand ) {
 			this.listenTo( enterCommand, 'afterExecute', ( evt, data ) => {
 				const positionParent = editor.document.selection.getFirstPosition().parent;
 				const batch = data.batch;
-				const isHeading = formats.some( ( format ) => format.id == positionParent.name );
+				const isHeading = options.some( option => option.id == positionParent.name );
 
-				if ( isHeading && positionParent.name != command.defaultFormat.id && positionParent.childCount === 0 ) {
-					batch.rename( positionParent, command.defaultFormat.id );
+				if ( isHeading && positionParent.name != command.defaultOption.id && positionParent.childCount === 0 ) {
+					batch.rename( positionParent, command.defaultOption.id );
 				}
 			} );
 		}
 	}
+
+	/**
+	 * Returns heading options as defined in `config.heading.options` but processed to consider
+	 * editor localization, i.e. to display {@link module:heading/headingcommand~HeadingOption#label}
+	 * in the correct language.
+	 *
+	 * Note: The reason behind this method is that there's no way to use {@link utils/locale~Locale#t}
+	 * when the user config is defined because the editor does not exist yet.
+	 *
+	 * @private
+	 * @returns {Array.<module:heading/headingcommand~HeadingOption>}.
+	 */
+	_getLocalizedOptions() {
+		if ( this._cachedLocalizedOptions ) {
+			return this._cachedLocalizedOptions;
+		}
+
+		const editor = this.editor;
+		const t = editor.t;
+		const localizedLabels = {
+			Paragraph: t( 'Paragraph' ),
+			'Heading 1': t( 'Heading 1' ),
+			'Heading 2': t( 'Heading 2' ),
+			'Heading 3': t( 'Heading 3' )
+		};
+
+		/**
+		 * Cached localized version of `config.heading.options` generated by
+		 * {@link module:heading/headingengine~HeadingEngine#_localizedOptions}.
+		 *
+		 * @private
+		 * @readonly
+		 * @member {Array.<module:heading/headingcommand~HeadingOption>} #_cachedLocalizedOptions
+		 */
+		return ( this._cachedLocalizedOptions = editor.config.get( 'heading.options' )
```

Comment:
Srsly, a multiline assignment combined with a return statement? Please, de-obfuscate this :P